irods/test/user_group_test.py

```diff
@@ -336,6 +338,84 @@ def test_add_user_metadata(self):
+                               port = self.sess.port,
+                               zone = self.sess.zone)
+
+        # Return two objects: the user and the session.
+        return (groupadmin, session)
+
+    def test_group_admin_can_create_and_administer_groups__issue_426(self):
+        admin = self.sess
+        alice = groupadmin = lab = None
+        GROUP_ADMIN = 'groupadmin_426'
+        try:
+            # Create a rodsuser and a groupadmin.
+            alice = admin.users.create('alice','rodsuser')
+            alice.modify('password', 'apass')
+            groupadmin, groupadmin_session = self.create_groupadmin_user_and_session(GROUP_ADMIN)
+
+            # As the groupadmin:
+            #    * Add two users to the group (one being the groupadmin) and assert membership.
+            #    * Remove those users from the group, and assert they are no longer members.
+            with groupadmin_session:
+                lab = groupadmin_session.groups.create('lab')
+                groupadmin_session.groups.addmember('lab',GROUP_ADMIN)
+                groupadmin_session.groups.addmember('lab','alice')
+
+                # Check that our members got added.
+                # (For set objects in Python, s1 <= s2 comparison means "is s1 a subset of s2?")
+                self.assertLessEqual(set(('alice',GROUP_ADMIN)), set(member.name for member in lab.members))
+
+                groupadmin_session.groups.removemember('lab','alice')
+                groupadmin_session.groups.removemember('lab',GROUP_ADMIN)
+
+                # Check that our members got removed.
+                self.assertFalse(set(('alice',GROUP_ADMIN)) & set(member.name for member in lab.members))
+        finally:
+            if groupadmin:
+                groupadmin.remove()
+            # NB: groups and users, even if created by a groupadmin, must be removed by a rodsadmin.
+            if alice:
+                alice.remove()
+            if lab:
+                admin.groups.remove(lab.name)
+
+    def test_group_admin_can_create_users__issue_428(self):
+        admin = self.sess
+        if admin.server_version < (4, 2, 12) or admin.server_version == (4,3,0):
+            self.skipTest('Password initialization is broken before iRODS 4.2.12, and in 4.3.0')
+        bob = groupadmin = None
+        bobs_password = 'random_password'
+        try:
+            # Create a groupadmin.
+            groupadmin, groupadmin_session = self.create_groupadmin_user_and_session('groupadmin_428')
+
+            # Use the groupadmin to create a new user initialized with a known password; then, test the
+            # new user/password combination by logging in and grabbing the home collection object.
+            with groupadmin_session:
+                bob = groupadmin_session.users.create_with_password('bob', bobs_password)
+                with iRODSSession(user = 'bob', password = bobs_password, host = admin.host, port = admin.port, zone = admin.zone) as bobs_session:
```

Comment:
Consider assigning `'bob'` to a variable to protect the test from future modifications.

Reply:
Yes, done.